### PR TITLE
Angular mocha testing

### PR DESCRIPTION
**Pull request by KorinaGraber**
Updated to angular rc 4. Simplified tests with a lot of dependencies using the angular test tooling to specify that we're using utility providers.

Fixed bugs with the angular 2 simple cards and long click button caught when regression testing the angular 2 components. 
In the long click button, failed long clicks would log an error to the console. Addressed by modifying the logic so the timeout catch handler fires the warning message. This is the intended purpose for why I made the timeout throw on a cancel.
In simple cards, a naming conflict with the base class was resulting in an infinite loop when the card tried to close.

### CODE REVIEW

**Comment by KorinaGraber** [comment]
Verified in regression testing that the forms still work in the new release candidate

**Comment by KorinaGraber** [comment]
The other thing we may want to look into at some point is the angular tooling for building components with their html. That might let us actually unit test the view for some key components.

**Comment by JoshDSommer** [comment]
:shipit:

File: source/components/simpleCardList/simpleCard.ts [hunk 1]
```diff
@@ -81,10 +81,6 @@ export class SimpleCardComponent<T> extends FormComponent implements OnInit {
 		return canClose;
 	}
 
-	saveForm(): boolean {
-		return this.boolean.toBool(this.submit());
```
Comment (JamieBowen) [inline source/components/simpleCardList/simpleCard.ts:85]: It looks like the boolean service is no longer being used so the import statement and DI can be removed as well.

File: source/components/buttons/buttonLongClick/buttonLongClick.ts [hunk 2]
```diff
@@ -43,15 +43,11 @@ export class ButtonLongClickComponent extends ButtonAsyncComponent {
 		this.timer = this.timeoutService.setTimeout((): void => {
 			this.cleanup();
 			this.triggerAction($event);
-		}, this.duration);
+		}, this.duration).catch(() => this.warn());
 	}
```
Comment (JamieBowen) [inline source/components/buttons/buttonLongClick/buttonLongClick.ts:47]: Nice improvement to eliminate the console errors

File: source/components/buttons/buttonLongClick/buttonLongClick.ts [hunk 3]
```diff
@@ -43,15 +43,11 @@ export class ButtonLongClickComponent extends ButtonAsyncComponent {
...
 
 	stopAction(): void {
-		if (this.active) {
-			if (this.timer != null) {
-				this.warn();
-			}
-
+		if (this.timer) {
```
Comment (JamieBowen) [inline source/components/buttons/buttonLongClick/buttonLongClick.ts:50]: Truthy FTW :+1:

File: source/components/simpleCardList/simpleCard.ts (hunk 1 above)
Comment (KorinaGraber) [inline source/components/simpleCardList/simpleCard.ts:85]: :+1:

Comment (KorinaGraber) [inline source/components/simpleCardList/simpleCard.ts:85]: https://github.com/RenovoSolutions/TypeScript-Angular-Components/pull/316/commits/96f60d44fd376be85d1ac3e5170420c7827ab672

**Comment by JamieBowen** [comment]
:shipit:

**Pull request merged**